test/cypress/support/secret_providers.ts

```diff
@@ -0,0 +1,37 @@
+import type { GetCypressChainableFromCommands } from 'cloudify-ui-common-cypress/support';
+import { addCommands } from 'cloudify-ui-common-cypress/support';
+
+declare global {
+    namespace Cypress {
+        // NOTE: necessary for extending the Cypress API
+        // eslint-disable-next-line @typescript-eslint/no-empty-interface
+        export interface Chainable extends GetCypressChainableFromCommands<typeof commands> {}
+    }
+}
+export interface SecretProvider {
+    name: string;
+    type: string;
+    visibility: string;
+}
+
+const commands = {
+    createSecretProvider: (secretProvider: SecretProvider) => {
+        return cy.cfyRequest('/secrets-providers', 'PUT', null, {
+            name: secretProvider.name,
+            type: secretProvider.type,
+            visibility: secretProvider.visibility
+        });
```

Comment:
WDYT about a bit shorter version: 
```suggestion
        const { name, type, visibility } = secretProvider;
        return cy.cfyRequest('/secrets-providers', 'PUT', null, { name, type, visibility });
```
? 🙂